Rewrite whatnow.xml

tox-wtf opened the pull request:
Hi,

I noticed areas for improvement for the whatnow page. This PR fixes a technical inaccuracy regarding i3[^1], any typos I noticed, as well as some grammatical errors.

I also attempted to restructure and rephrase sentences, paragraphs, and sections for clarity. The biggest restructure was desktop environments; I split them into their own section, and restructured the components of a graphical environment in a way that I think makes more sense. This is very much in a draft state, and I included some commented questions asking for your input. I'm also open to any feedback or requested changes you have.

Thanks!

[^1]: i3 is not supported on Wayland.

Commit:
rewrote whatnow

fixed some technical inaccuracies, typos, and grammatical errors. rephrased several sentences for clarity. broke desktop environments into their own section. added comments.

zeckma-tech commented on shareddeps/checkpoint/whatnow.xml:
```diff
@@ -12,31 +12,30 @@
-    hardest part of this book. Congrats!
+    By now, you should have most of the packages needed to compile any packages
+    mentioned on this page. The good news is you are now done with the hardest
+    part of this book. Congrats!
```
I still think that "the following packages" is a better fit, or "packages that come after this page." Because there are more packages than just Steam, Wine, and DXVK, such as all the dependencies needed for Wine, and the recommended deps for Steam.

Toxikuu replied:
Should I change it to "the packages on this page and in the next sections"? I'm unsure whether "the following packages after this section" means the following packages on this page, or the packages after this section (or both).

tox-wtf replied:
Should I go with "packages on this page and beyond"?

tox-wtf replied:
I went with the above phrasing. I was originally unsure if you meant the packages mentioned on this page, after, or both. But I think this captures the intended meaning(?)

zeckma-tech commented on shareddeps/checkpoint/whatnow.xml:
```diff
@@ -12,31 +12,30 @@
+    and not accidentally lock yourself inside of it without a hard reboot. The
+    installation of those components is not covered in this book, as the
+    specific components will depend on if you plan to use
+    <application>X11</application>, <application>Wayland</application>, or both.
```
Line goes over 80 characters.

tox-wtf replied:
Several lines went over 80 characters before I made any edits. I took this to mean I should balance perceived readability in the xml with the 80 character limit. Should I more strictly adhere to the 80 character limit?

tox-wtf replied:
I went with more strictly adhering to the 80 character limit.

zeckma-tech commented on shareddeps/checkpoint/whatnow.xml:
```diff
@@ -12,31 +12,30 @@
-    While you are at it, make sure you have followed
-    <xref linkend="bash-profile"/> if you have not already as the files in that
-    section set necessary variables.
+    While you are at it, make sure you have followed <xref linkend="bash-profile"/>
```
Line goes over 80 characters.

zeckma-tech commented on shareddeps/checkpoint/whatnow.xml:
```diff
@@ -168,52 +170,35 @@
-          be loaded by a display manager, like <ulink url="&blfs-svn;/x/sddm.html">
-          SDDM</ulink>. A display manager is often loaded by a service or daemon.
-        </para>
-
```
No, please do not remove the desktop environment content. It could be valuable information to many people. And I do not see an issue with how it is right now.

tox-wtf replied:
I attempted retain all information regarding desktop environments. I just reorganized it in a way I find is clearer. The page currently mixes desktop environments with the three components. I think structuring the page to explain the components, then explaining how desktop environments bundle those components together is a lot clearer than having the desktop environment sprinkled in throughout. 

If you'd still like me to change it back to the way it was, I'm fine with doing so. Just let me know if that's the case.

zeckma-tech replied:
I want it back to the way it was.

zeckma-tech commented on shareddeps/checkpoint/whatnow.xml:
```diff
@@ -72,15 +71,16 @@
         </para>
       </important>
 
+      <!-- Should using xrandr to set a higher refresh rate also be covered? -->
```
It's covered in the BLFS link at the top of the page.

zeckma-tech commented on shareddeps/checkpoint/whatnow.xml:
```diff
@@ -110,10 +110,11 @@
-        GPU specifically, like Nouveau for NVIDIA GPUs. You can override this
-        by setting the following variable:
+        If you compiled Zink in <xref linkend="mesa"/>, chances are
+        <application>X11</application> or any <application>Wayland</application>
```
Line goes over 80 characters.

zeckma-tech commented on shareddeps/checkpoint/whatnow.xml:
```diff
@@ -146,18 +148,18 @@
-          window managers include <ulink url="https://awesomewm.org/">AwesomeWM</ulink>,
+          A window manager is an application that manages windows, their
+          placement, how they move, and more. A window manager is a minimal
+          option for a graphical environment. Some popular window managers include
```
Line goes over 80 characters.

zeckma-tech commented on shareddeps/checkpoint/whatnow.xml:
```diff
@@ -168,52 +170,35 @@
-          terminals you can go for and some on the page are in BLFS. If you decided
-          to go with a window manager, make sure to add a keybind for your chosen
-          terminal.
+          Category:Terminal emulators</ulink> for a comprehensive list of terminals.
```
Line goes over 80 characters.

zeckma-tech commented on shareddeps/checkpoint/whatnow.xml:
```diff
@@ -168,52 +170,35 @@
-          Category:Application launchers</ulink> for a comprehensive list on some
-          launchers you can go for. Desktop environments include their own
-          launcher and are often accessed via the <quote>start menu</quote>.
-          If you went with a window manager, make sure to add the keybind for the launcher.
```
Please do not remove the bit about how launchers can be accessed via the start menu in a desktop environment.

tox-wtf replied:
It's moved, not removed.

zeckma-tech commented on shareddeps/checkpoint/whatnow.xml:
```diff
@@ -223,26 +208,52 @@
         <para>
-          Desktop environments include their own and are started when the
-          desktop environment loads up.
+          Note that some window managers handle compositing on their own.
```
Include desktop environments in the mix.

tox-wtf replied:
Here I wanted to point out the detail that some window managers handle compositing on their own, as discussed in Section 6 of https://wiki.archlinux.org/title/Xorg#Composite.

I discussed desktop environments after (reference the first unresolved request for rationale).

zeckma-tech replied:
I see, my bad.

zeckma-tech commented on shareddeps/checkpoint/whatnow.xml:
```diff
@@ -223,26 +208,52 @@
+          start a terminal or launcher is less important on a desktop
+          environment than on a window manager, since you usually have access to
+          a start menu with just the mouse.
         </para>
```
Just include this in the same area as window managers.

zeckma-tech commented on shareddeps/checkpoint/whatnow.xml:
```diff
@@ -255,34 +266,36 @@
 EOF</userinput></screen>
 
+      <!-- Maybe consider restructuring here as 'Wayland compositor' is
+      referenced before it is defined? -->
```
We can then just change "compositor" to whatever without restructuring anything.

tox-wtf replied:
I'll address this in the next commit. Until then, I'll keep this unresolved so that I remember to do so.

tox-wtf replied:
Addressed by changing "compositor" to "chosen graphical environment".

zeckma-tech commented on shareddeps/checkpoint/whatnow.xml:
```diff
@@ -255,34 +266,36 @@
+        applicable to you. You won't need to worry about making sure the
+        <application>NVIDIA</application> driver is used when starting
         <application>Wayland</application>. However, read through the
         installation instructions for compositors as they will likely have
```
Read comment above.

zeckma-tech commented on shareddeps/checkpoint/whatnow.xml:
```diff
@@ -291,23 +304,22 @@ EOF</userinput></screen>
       <sect4>
         <title>Compositors</title>
 
+        <!-- Consider excluding this paragraph as it seems mostly irrelevant? -->
```
It's an important difference that users should probably know.

Commit:
complied with some requested changes

more strictly adhered to the 80 character limit. removed comments. phrasing at the start

tox-wtf commented:
I've complied with some of the requested changes.

I'll unrestructure the components section if you wish, but would like confirmation before I do so.

Zeckmathederg committed:
Keep up with BLFS.

Cherry picked from commit 06470825344b81d5838eb611f925b9bfb7be5403 to
include changes in stylesheets/lfs-xsl/xhtml which provide footnote
support.

Commit:
Added footnotes and remaining requested changes

Undid the desktop environment restructuring

tox-wtf commented:
I believe that's everything. Please look through it and let me know if there's anything I missed. Thanks!

Zeckmathederg committed:
Merge branch 'trunk' into whatnow-rewrite

Zeckmathederg reviewed (approved):
Okay, this looks good to me now. Merging, and many thanks for working with me.
